src/QMCWaveFunctions/TrialWaveFunction.cpp

```diff
@@ -1350,17 +1378,35 @@ void TrialWaveFunction::initializeTWFFastDerivWrapper(const ParticleSet& P, TWFF
   {
     if (Z[i]->isFermionic())
     {
-      //OK, so this is a hack only for SlaterDeterminant objects.
-      //Needs a bit of logic and protection before this reaches production.
-      //SlaterDet* det = dynamic_cast<SlaterDet*>(Z[i].get());
-      //det->registerTWFFastDerivWrapper(P, twf);
       Z[i]->registerTWFFastDerivWrapper(P, twf);
     }
     else
       twf.addJastrow(Z[i].get());
   }
 }
 
+
+// Add debug checks in getOrCreateTWFFastDerivWrapper
+TWFFastDerivWrapper& TrialWaveFunction::getOrCreateTWFFastDerivWrapper(const ParticleSet& P)
+{
+  if (!twf_fastderiv_)
+  {
+    twf_fastderiv_             = std::make_unique<TWFFastDerivWrapper>();
+    twf_fastderiv_last_P_      = nullptr;
+    twf_fastderiv_last_Zcount_ = 0;
+  }
+
+  const bool need_init = (twf_fastderiv_last_P_ != &P) || (twf_fastderiv_last_Zcount_ != Z.size());
+  if (need_init)
+  {
+    initializeTWFFastDerivWrapper(P, *twf_fastderiv_);
+    twf_fastderiv_last_P_      = &P;
+    twf_fastderiv_last_Zcount_ = Z.size();
+  }
```

Comment:
Based on my understanding of the TWFFastDerivWrapper.
Even if `twf_fastderiv_last_P_ != &P` there is no need to re-initialize the wrapper.
`twf_fastderiv_last_Zcount_ != Z.size()` probably can only detect if `twf_fastderiv_` has been initialized.
So why not just call `initializeTWFFastDerivWrapper` inside `if (!twf_fastderiv_)`?

Reply:
Strange... I pushed it on my side and it shows on my branch but not here...